PatternPal/PatternPal.Extension/Views/StepByStepInstructionsView.xaml.cs

```diff
@@ -104,67 +107,92 @@ private void InitializeViewModelAndButtons()
 
                                       _viewModel = (StepByStepInstructionsViewModel)(this.DataContext);
 
+                                      if (!_viewModel.SBScontinue)
+                                      {
+                                          ClassSelection.Visibility = Visibility.Hidden;
+                                      }
+
                                       CheckIfNextPreviousButtonsAvailable();
                                   };
         }
 
+        /// <summary>
+        /// Checks if the check implementation button is able to be clicked
+        /// </summary>
         private void CheckIfCheckIsAvailable()
         {
-            ClassSelection.Visibility = _viewModel.CurrentInstruction.ShowFileSelector
-                ? Visibility.Visible
-                : Visibility.Hidden;
-            CheckImplementationButton.IsEnabled = !_viewModel.CurrentInstruction.ShowFileSelector || ClassSelection.SelectedItem != null;
-
+            if (_viewModel.SBScontinue)
+            {
+                CheckImplementationButton.IsEnabled = ClassSelection.SelectedItem != null;
+            }
+            else
+            {
+                CheckImplementationButton.IsEnabled = 
+                    _viewModel.CurrentInstruction.FileId != "" || !_viewModel.SBScontinue;
```

Comment:
```suggestion
                    !string.IsNullOrEmpty(_viewModel.CurrentInstruction.FileId) || !_viewModel.SBScontinue;
```

Reply:
Checking if the check button should be visible is now done by checking whether the filepath(s) that is going to be used for a check call to the service is non empty.